filters/openpolicyagent/openpolicyagent.go

```diff
@@ -0,0 +1,318 @@
+package openpolicyagent
+
+import (
+	"bytes"
+	"context"
+	"os"
+	"strings"
+	"sync"
+	"text/template"
+	"time"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/config"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/discovery"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/runtime"
+	"github.com/open-policy-agent/opa/storage"
+	"github.com/open-policy-agent/opa/storage/inmem"
+	iCache "github.com/open-policy-agent/opa/topdown/cache"
+	opautil "github.com/open-policy-agent/opa/util"
+	"github.com/opentracing/opentracing-go"
+
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/envoy"
+	"github.com/zalando/skipper/filters/openpolicyagent/internal/util"
+)
+
+type OpenPolicyAgentFactory struct {
+	// Ideally share one Bundle storage across many OPA "instances".
+	// This allows to save memory on bundles that are shared
+	// between different policies (i.e. global team memberships)
+	// This not possible due to some limitations in OPA
+	// See https://github.com/open-policy-agent/opa/issues/5707
+	bundleStorage storage.Store
+}
+
+func NewOpenPolicyAgentFactory() OpenPolicyAgentFactory {
+	return OpenPolicyAgentFactory{
+		//BundleStorage: inmem.New(),
```

Comment:
?

Reply:
Yes, the issue that is required to put this back in is anyways linked a few lines up. Will remove